Add test-utils project

dlindhol opened the pull request.

dlindhol committed:
Add test-utils project

dlindhol commented on test-utils/src/test/scala/latis/util/DatasetTestUtilsSuite.scala:
```diff
@@ -0,0 +1,105 @@
+    assert(x.getMessage == "Empty dataset")
+  }
+
+  test("equals first sample") {
```
This could be really handy for our quick first sample tests. Just need an id and a list of raw values.

dlindhol commented on test-utils/src/main/scala/latis/util/DatasetTestUtils.scala:
```diff
@@ -0,0 +1,71 @@
+
+abstract class DatasetTestUtils extends AnyFunSuite {
+
+  val catalog: Catalog
```
I imagine each service like `latis3-lisird` would have a subclass of `DatasetTestUtils` that encapsulates the catalog generation from the fdml. Each dataset test would in turn extend it. Or maybe consolidate the first sample tests in a single class? Or just use a csv file with `id, values`?

dlindhol committed:
Remove JUnit

dlindhol committed:
Use DatasetGenerator instead of fdml

dlindhol committed:
Use user.dir so this test should work in other environments

dlindhol committed:
Ignore GranuleListJoin test

dlindhol committed:
Fix fs2 deprecations

dlindhol committed:
Suppress compiler warnings

dlindhol committed:
Address compiler warnings in tests

dlindhol commented on core/src/main/scala/latis/input/Adapter.scala:
```diff
@@ -13,6 +13,7 @@ trait Adapter {
   //TODO: add SmartAdapter that takes ops?
   //TODO: require UnaryOperations
 
+  @annotation.nowarn("msg=never used")
```
TODO

dlindhol replied:
categories or "unused" or use "_"?

dlindhol replied:
Scala doesn't like "_" for the parameter, but we can embed the annotation closer to the problem and use a category:
```
def canHandleOperation(@nowarn("cat=unused") op: Operation): Boolean = false
```

lindholc replied:
I guess `_` was a TypeScript thing.

It's probably better to use `@unused` if that works. It's shorter, at least.

dlindhol replied:
Their article didn't mention `@unused`. I'll change that.

dlindhol commented on core/src/test/scala/latis/dsl/ModelParserSpec.scala:
```diff
@@ -148,7 +149,7 @@ class ModelParserSpec extends AnyFlatSpec {
    * want to parse and the thing you expect to get back
    */
-  private def testParser[A](p: Parser[A])(s: String, d: A): Unit = p.parseOnly(s) match {
+  private def testParser[A](p: Parser[A])(s: String, d: A): Any = p.parseOnly(s) match {
```
refine?

dlindhol replied:
Some of these `Any`s have become `Assertion`.

dlindhol commented on test-utils/src/main/scala/latis/util/DatasetTestUtils.scala:
```diff
@@ -0,0 +1,71 @@
+import latis.model.DataType
+import latis.ops._
+
+abstract class DatasetTestUtils extends AnyFunSuite {
```
"Suite"?

dlindhol committed:
Rename DatasetTestSuite

dlindhol committed:
Address compile warnings

Many remaining warnings relate to exhaustivity of DataType (#304)
and Data matches (#305).

dlindhol committed:
Use unused annotation
